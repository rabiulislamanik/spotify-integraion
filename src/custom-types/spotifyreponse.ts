export interface ISpotifyCommonResponse {
    next:string|null,
    total:number
}

export interface IAlbums extends ISpotifyCommonResponse{
    items:IAlbumItem[]
}
  
export interface INewReleaseAlbumResponse{
    albums:IAlbums
}

export interface IAlbumItem {
    album_type:string,
    artists:{external_urls:{spotify:string},name:string}[],
    external_urls:{ spotify: string},
    name:string,
    release_date:string,
    total_tracks:number,
    images:{url:string}[]
}

export interface IPlaylists extends ISpotifyCommonResponse{
    items:IPlaylistItem[]
}

export interface IFeaturedPlaylistsResponse{
    message:string,
    playlists:IPlaylists
}


export interface IPlaylistItem {
    description:string,
    external_urls:{ spotify: string},
    name:string,
    owner:{display_name:string,external_urls:{ spotify: string}},
    images:{url:string}[]
}


export interface ICategories extends ISpotifyCommonResponse{
    items:ICategoryItem[]
}

export interface ICategoriesResponse{
    message:string,
    categories:ICategories
}

export interface ICategoryItem {
    icons:{url:string}[],
    name:string,
    id:string
}
