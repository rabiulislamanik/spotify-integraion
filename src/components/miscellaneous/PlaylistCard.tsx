import React from "react";
import { IPlaylistItem } from '../../custom-types/spotifyreponse';

interface IPlaylistCard{
    playlist:IPlaylistItem
}

function PlaylistCard (props:IPlaylistCard){
  return (
    <div className="card">
      <img className="card-image" src={props.playlist.images[0].url} alt="icon" height="128" width="128"/>
      <p><a className="card-link" href={props.playlist.external_urls.spotify}>{props.playlist.name}</a></p>
      <p>{props.playlist.description}</p>
      <p>Playlist by : <a className="card-link" href={props.playlist.owner.external_urls.spotify}>
          {props.playlist.owner.display_name}
      </a></p>
    </div>
  );
}

export default PlaylistCard;