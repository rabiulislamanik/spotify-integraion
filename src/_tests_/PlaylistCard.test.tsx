import React from 'react';
import renderer from 'react-test-renderer';

import PlaylistCard from '../components/miscellaneous/PlaylistCard';
import { IPlaylistItem } from '../custom-types/spotifyreponse';

it('renders correctly when an Playlist is passed as prop', () => {
  const fakePlaylist:IPlaylistItem = {
    description:'Title',
    external_urls:{ spotify: 'Url_to_spotify'},
    name:'Dummy Playlist',
    owner:{display_name:'Dummy Owner',external_urls:{ spotify: 'Another url to spotify'}},
    images:[{url:'url_to_image'}]
  }; 
  const tree = renderer.create(<PlaylistCard playlist={fakePlaylist}/>).toJSON();
  expect(tree).toMatchSnapshot();
});