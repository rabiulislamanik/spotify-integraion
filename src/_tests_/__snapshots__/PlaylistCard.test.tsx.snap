// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when an Playlist is passed as prop 1`] = `
<div
  className="card"
>
  <img
    alt="icon"
    className="card-image"
    height="128"
    src="url_to_image"
    width="128"
  />
  <p>
    <a
      className="card-link"
      href="Url_to_spotify"
    >
      Dummy Playlist
    </a>
  </p>
  <p>
    Title
  </p>
  <p>
    Playlist by : 
    <a
      className="card-link"
      href="Another url to spotify"
    >
      Dummy Owner
    </a>
  </p>
</div>
`;
